src/py21cmsense/observation.py

```diff
@@ -94,16 +94,16 @@ class Observation:
 
     time_per_day: tp.Time = attr.ib(
         6 * un.hour,
-        validator=(tp.vld_physical_type("time"), ut.between(0 * un.hour, 24 * un.hour)),
+        validator=(tp.vld_physical_type("time"), ut.between(0 * un.hour, 655.2 * un.hour)),
```

Comment:
Instead of changing the validator here, you could use a more dynamic validator. To do this, remove the `ut.between()` validator from this tuple, and add the following:

```python
@time_per_day.validator
def _time_per_day_vld(self, att, val):
    day_length = 24*un.hour if self.observatory.world == 'earth' else 655.2*un.hour

    if not 0*un.hour <= val <= day_length:
        raise ValueError(f"time_per_day should be between 0 and {day_length}")
```

Note that since the `observatory` attribute is defined before the `time_per_day` attribute on the `Observation` class, it will be available at validation time. Use this same pattern for the other attributes that require adjusting in the PR.